tst/EnergyPlus/unit/OutputProcessor.unit.cc

```diff
@@ -5350,6 +5352,138 @@ namespace OutputProcessor {
+                    state->dataEnvrn->DayOfWeek = 1;
+                }
+
+                state->dataGlobal->EndDayFlag = false;
+                for (state->dataGlobal->HourOfDay = 1; state->dataGlobal->HourOfDay <= 24; ++state->dataGlobal->HourOfDay) {
+                    for (state->dataGlobal->TimeStep = 1; state->dataGlobal->TimeStep <= state->dataGlobal->NumOfTimeStepInHour;
+                         ++state->dataGlobal->TimeStep) {
+                        if (state->dataGlobal->TimeStep == state->dataGlobal->NumOfTimeStepInHour) {
+                            state->dataGlobal->EndHourFlag = true;
+                            if (state->dataGlobal->HourOfDay == 24) {
+                                state->dataGlobal->EndDayFlag = true;
+                                if ((!state->dataGlobal->WarmupFlag) && (state->dataGlobal->DayOfSim == state->dataGlobal->NumOfDayInEnvrn)) {
+                                    state->dataGlobal->EndEnvrnFlag = true;
+                                }
+                            }
+                        }
+
+                        if (state->dataEnvrn->DayOfMonth == state->dataWeather->EndDayOfMonth(state->dataEnvrn->Month)) {
+                            state->dataEnvrn->EndMonthFlag = true;
+                        }
+                        records_written += numOutputVariables;
+                        if (records_written > (INT_MAX_AS_SIZE_T - numOutputVariables)) {
+                            EXPECT_EQ("2005-12-22 02:45",
+                                      fmt::format("{:04d}-{:02d}-{:02d} {:02d}:{:02d}",
+                                                  state->dataGlobal->CalendarYear,
+                                                  state->dataEnvrn->Month,
+                                                  state->dataEnvrn->DayOfMonth,
+                                                  state->dataGlobal->HourOfDay,
+                                                  state->dataGlobal->TimeStep));
+                            UpdateMeterReporting(*state);
+                            UpdateDataandReport(*state, TimeStepType::Zone);
+                            found_about_to_overflow = true;
+                            break;
+                        }
+                    }
+                    if (found_about_to_overflow) {
+                        break;
+                    }
+                }
+                if (found_about_to_overflow) {
+                    break;
+                }
+            }
+            if (found_about_to_overflow) {
+                break;
+            }
+        }
+        ASSERT_TRUE(found_about_to_overflow);
+        EXPECT_EQ(numOutputVariables + 1, state->dataGlobal->StdOutputRecordCount);
+        EXPECT_EQ(1, state->dataGlobal->StdMeterRecordCount);
+        EXPECT_TRUE(has_eso_output(true));
+        state->dataGlobal->StdOutputRecordCount = records_written;
+        state->dataGlobal->StdMeterRecordCount = INT_MAX;
```

Comment:
Fake having written the record for all timesteps until it overflows. if I don't fake it, it's unbearably slow